### PR TITLE
Bulk registration

### LINKED ISSUE #2112: Make item registration a bulk action
**Input**

This will take as an input a spreadsheet with all of the data fields that are submitted from the registration form. Each row will represent the data for one druid to be created.

**Output**

For the output, there should be a report listing the druids created, similar to the output that appears in the grid section of the registration form. This should be downloadable like the report generated from the checksum bulk action.

**Data**

Each row should have headers for:

APO (required)
Collection
Rights (required)
Initial Workflow (required)
Content Type (required)
Project Name
Tags (not required, may repeat)
Catkey
Source ID (required)
Label (required if no Catkey)

**Desired behavior if a single item (row) fails**

Ideally, the registration process should continue if there is a problem with an item, for example if one of the source_ids already exists for a previously created object. This should be listed in the log.

**Additional context**

In addition to scaling better, submitting from a spreadsheet as a bulk action will allow people to register batches that are not limited to a single combination of APO, collection, rights, tags, etc. per batch.

The one bit of functionality (besides the UI) that this bulk action may not port over from the form is the ability to print tracking sheets. I'm not sure how we could carry that over.

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?

Fixes #2112 

## How was this change tested?
Tested on staging.

## Which documentation and/or configurations were updated?
n/a

### CODE REVIEW

**Comment by ndushay** [comment]
is this signed off on by Andrew?  might be good to make a note of it here.

**Comment by andrewjbtw** [comment]
I have a few comments on the issue thread: https://github.com/sul-dlss/argo/issues/2112

The main one is I haven't been able to assign multiple tags at registration beyond 1 project tag and one "ordinary" tag.

**Comment by jcoyne** [comment]
@andrewjbtw I've fixed the problem with the multiple tags, can you please retest that?

**Comment by andrewjbtw** [comment]
@jcoyne I was just able to register an item with multiple tags, but got errors on the items that didn't have multiple tags.

```
2020-08-17 21:46:26 UTC Starting RegisterDruidsJob for BulkAction 564
2020-08-17 21:46:26 UTC RegisterDruidsJob: Registering with {:model=>#<Cocina::Models::RequestDRO type="http://cocina.sul.stanford.edu/models/object.jsonld" label="bulk_36" version=1 access=#<#<Class:0x000000000930e658> access="world" copyright=nil embargo=nil download="world" readLocation=nil useAndReproductionStatement=nil> administrative=#<Cocina::Models::Administrative hasAdminPolicy="druid:sc012gz0974" releaseTags=nil partOfProject="bulk registration"> description=nil identification=#<Cocina::Models::RequestIdentification sourceId="bulk:bulk_36" catalogLinks=[#<Cocina::Models::CatalogLink catalog="symphony" catalogRecordId="444">]> structural=#<#<Class:0x000000000a12f548> contains=nil hasMemberOrders=nil isMemberOf="druid:bk024qs1808" hasAgreement=nil> geographic=nil>, :workflow=>"registrationWF", :tags=>["csv : test", "Project : two", nil]}
2020-08-17 21:46:33 UTC RegisterDruidsJob: Bad Request: 400 ({"errors":[{"status":"bad_request","detail":"#/components/schemas/AdministrativeTag does not allow null values"}]}) for druid:hy468jm6660
2020-08-17 21:46:33 UTC RegisterDruidsJob: Registering with {:model=>#<Cocina::Models::RequestDRO type="http://cocina.sul.stanford.edu/models/object.jsonld" label="bulk_37" version=1 access=#<#<Class:0x000000000930e658> access="world" copyright=nil embargo=nil download="none" readLocation=nil useAndReproductionStatement=nil> administrative=#<Cocina::Models::Administrative hasAdminPolicy="druid:sc012gz0974" releaseTags=nil partOfProject="bulk registration"> description=nil identification=#<Cocina::Models::RequestIdentification sourceId="bulk:bulk_37" catalogLinks=[#<Cocina::Models::CatalogLink catalog="symphony" catalogRecordId="777">]> structural=#<#<Class:0x000000000a12f548> contains=nil hasMemberOrders=nil isMemberOf=nil hasAgreement=nil> geographic=nil>, :workflow=>"registrationWF", :tags=>["csv : test", "Project : two", "additional : tag"]}
2020-08-17 21:46:34 UTC RegisterDruidsJob: Successfully registered druid:ww393np5012
2020-08-17 21:46:34 UTC RegisterDruidsJob: Registering with {:model=>#<Cocina::Models::RequestDRO type="http://cocina.sul.stanford.edu/models/object.jsonld" label="bulk_38" version=1 access=#<#<Class:0x000000000930e658> access="stanford" copyright=nil embargo=nil download="stanford" readLocation=nil useAndReproductionStatement=nil> administrative=#<Cocina::Models::Administrative hasAdminPolicy="druid:sc012gz0974" releaseTags=nil partOfProject="bulk registration"> description=nil identification=#<Cocina::Models::RequestIdentification sourceId="bulk:bulk_38" catalogLinks=[]> structural=#<#<Class:0x000000000a12f548> contains=nil hasMemberOrders=nil isMemberOf=nil hasAgreement=nil> geographic=nil>, :workflow=>"registrationWF", :tags=>["csv : test", "Project : two", nil]}
2020-08-17 21:46:35 UTC RegisterDruidsJob: Bad Request: 400 ({"errors":[{"status":"bad_request","detail":"#/components/schemas/AdministrativeTag does not allow null values"}]}) for druid:ct232nz0318
2020-08-17 21:46:35 UTC RegisterDruidsJob: Registering with {:model=>#<Cocina::Models::RequestDRO type="http://cocina.sul.stanford.edu/models/object.jsonld" label="bulk_39" version=1 access=#<#<Class:0x000000000930e658> access="stanford" copyright=nil embargo=nil download="none" readLocation=nil useAndReproductionStatement=nil> administrative=#<Cocina::Models::Administrative hasAdminPolicy="druid:sc012gz0974" releaseTags=nil partOfProject="bulk registration"> description=nil identification=#<Cocina::Models::RequestIdentification sourceId="bulk:bulk_39" catalogLinks=[#<Cocina::Models::CatalogLink catalog="symphony" catalogRecordId="999">]> structural=#<#<Class:0x000000000a12f548> contains=nil hasMemberOrders=nil isMemberOf=nil hasAgreement=nil> geographic=nil>, :workflow=>"registrationWF", :tags=>["csv : test", "Project : two", nil]}
2020-08-17 21:46:36 UTC RegisterDruidsJob: Bad Request: 400 ({"errors":[{"status":"bad_request","detail":"#/components/schemas/AdministrativeTag does not allow null values"}]}) for druid:zk735vb0462
2020-08-17 21:46:36 UTC RegisterDruidsJob: Registering with {:model=>#<Cocina::Models::RequestDRO type="http://cocina.sul.stanford.edu/models/object.jsonld" label="bulk_40" version=1 access=#<#<Class:0x000000000930e658> access="citation-only" copyright=nil embargo=nil download="none" readLocation=nil useAndReproductionStatement=nil> administrative=#<Cocina::Models::Administrative hasAdminPolicy="druid:sc012gz0974" releaseTags=nil partOfProject="bulk registration"> description=nil identification=#<Cocina::Models::RequestIdentification sourceId="bulk:bulk_40" catalogLinks=[]> structural=#<#<Class:0x000000000a12f548> contains=nil hasMemberOrders=nil isMemberOf=nil hasAgreement=nil> geographic=nil>, :workflow=>"registrationWF", :tags=>["csv : test", "Project : two", nil]}
2020-08-17 21:46:37 UTC RegisterDruidsJob: Bad Request: 400 ({"errors":[{"status":"bad_request","detail":"#/components/schemas/AdministrativeTag does not allow null values"}]}) for druid:tp514dd1227
2020-08-17 21:46:37 UTC RegisterDruidsJob: Registering with {:model=>#<Cocina::Models::RequestDRO type="http://cocina.sul.stanford.edu/models/object.jsonld" label="bulk_41" version=1 access=#<#<Class:0x000000000930e658> access="dark" copyright=nil embargo=nil download="none" readLocation=nil useAndReproductionStatement=nil> administrative=#<Cocina::Models::Administrative hasAdminPolicy="druid:sc012gz0974" releaseTags=nil partOfProject="bulk registration"> description=nil identification=#<Cocina::Models::RequestIdentification sourceId="bulk:bulk_41" catalogLinks=[]> structural=#<#<Class:0x000000000a12f548> contains=nil hasMemberOrders=nil isMemberOf=nil hasAgreement=nil> geographic=nil>, :workflow=>"registrationWF", :tags=>["csv : test", "Project : two", nil]}
2020-08-17 21:46:38 UTC RegisterDruidsJob: Bad Request: 400 ({"errors":[{"status":"bad_request","detail":"#/components/schemas/AdministrativeTag does not allow null values"}]}) for druid:pq771mt6634
2020-08-17 21:46:38 UTC RegisterDruidsJob: Registering with {:model=>#<Cocina::Models::RequestDRO type="http://cocina.sul.stanford.edu/models/3d.jsonld" label="bulk_42" version=1 access=#<#<Class:0x000000000930e658> access="location-based" copyright=nil embargo=nil download="location-based" readLocation="spec" useAndReproductionStatement=nil> administrative=#<Cocina::Models::Administrative hasAdminPolicy="druid:sc012gz0974" releaseTags=nil partOfProject="bulk registration"> description=nil identification=#<Cocina::Models::RequestIdentification sourceId="bulk:bulk_42" catalogLinks=[]> structural=#<#<Class:0x000000000a12f548> contains=nil hasMemberOrders=nil isMemberOf=nil hasAgreement=nil> geographic=nil>, :workflow=>"registrationWF", :tags=>["csv : test", "Project : two", nil]}
2020-08-17 21:46:39 UTC RegisterDruidsJob: Bad Request: 400 ({"errors":[{"status":"bad_request","detail":"#/components/schemas/AdministrativeTag does not allow null values"}]}) for druid:wx443qf5987
2020-08-17 21:46:39 UTC Finished RegisterDruidsJob for BulkAction 564
```

**Comment by andrewjbtw** [comment]
Looks like my CSV ended up with trailing commas. Let me try again.

**Comment by andrewjbtw** [comment]
Got the same result without trailing commas. I think having the column header for the additional tag may be why the 'nil' value is getting passed.

**Comment by jcoyne** [comment]
@andrewjbtw I just deployed a fix for the nil values.  See if that is better.

**Comment by andrewjbtw** [comment]
@jcoyne looks good - success on all druids. Tested with up to two tags beyond the project tags.

File: app/javascript/controllers/bulk_actions.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import { Controller } from 'stimulus'
 
 export default class extends Controller {
+  static targets = [ "commonFields" ]
```
Comment (aaron-collier) [inline app/javascript/controllers/bulk_actions.js:4]: is this used?

Comment (jcoyne) [reply app/javascript/controllers/bulk_actions.js:4]: Ah, good catch. It now is.
